### PR TITLE
Switch Travis to use Ubuntu 16.04 "xenial", and other CI tweaks

### PR DESCRIPTION
**Pull request by fingolfin**
This should allow us to build semigroups and digraphs again (alas, it seems digraphs still runs into these weird atexit troubles...)

### CODE REVIEW

**Comment by markuspf** [comment]
Not sure whether it makes sense; for `meataxe64` I am now also using `binutils-2.26`: (https://github.com/gap-packages/meataxe64/blob/master/.travis.yml)

I spent *quite* a bit of time trying to debug a linker problem with `meataxe64` which boiled down to "outdated binutils".

**Comment by markuspf** [comment]
sorry, I merged #2702 which caused a conflict with this PR...

**Comment by alex-konovalov** [comment]
gcc at least 5.0 would be very useful. The list of hacks to disable packages might now be extended with Digraphs (https://github.com/gap-system/gap/pull/2768). Hope with this PR all those hacks will be gone. But test of this PR currently fail to build packages:
```
checking whether GMP C++ library is available... configure: error: GMP C++ library not found. Make sure it was compiled with --enable-cxx
```

**Comment by alex-konovalov** [comment]
@fingolfin did you manage to get at least the gcc-5 of this PR working? I'd really love to have it because otherwise Semigroups can not be compiled on Travis, which severely reduces ability to identify potential problems.

**Comment by fingolfin** [comment]
This seems to be working now, at least with reduced tests. I've now cleaned up the PR, and in particular re-enabled all tests, so let's see if it still passes.

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+# Force Ubuntu 16.04 "Xenial" to get newer GCC, binutils etc.
+dist: xenial
 language: c
 
 env:
   global:
     - CFLAGS="--coverage -O2"
     - LDFLAGS="--coverage"
     # default config flags: enable debug asserts, and treat compiler warnings as errors
-    - CONFIGFLAGS="--enable-debug --enable-Werror"
```
Comment (alex-konovalov) [inline .travis.yml]: If we do not treat compiler warnings as errors (but why?) then the comment should be updated too. But I'd prefer to treat them as errors for the core GAP system, but not for packages.

Comment (fingolfin) [reply .travis.yml]: If you look at the commit messages, they explain that we actually keep using --enable-Werror, it simply is redundant to set this here again

Comment (fingolfin) [reply .travis.yml]: but yeah, the comment should be updated

File: .travis.yml [hunk 2]
```diff
@@ -58,7 +61,7 @@ matrix:
           - libcurl4-openssl-dev:i386 # for curlInterface
           - libboost-dev:i386       # for NormalizInterface
           - libmpfr-dev:i386        # for float
-          - libmpfi-dev:i386        # for float
+          - libmpfi0-dev:i386       # for float
```
Comment (markuspf) [inline .travis.yml]:
```suggestion
          - libmpfi0-dev:i386        # for float
```

**Pull request merged**